sdk/iot/hub/samples/src/generate_certificate.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+echo "Generating DEMO User Root Cert"
+echo "DO NOT USE THIS IN PRODUCTION"
+openssl ecparam -out user_root_cert_key.pem -name prime256v1 -genkey
+openssl req -new -days 365 -nodes -x509 -key user_root_cert_key.pem -out user_root_cert.pem -subj "/CN=CA Group 1"
+
+echo "Generating DEMO User Device Cert"
+echo "DO NOT USE THIS IN PRODUCTION"
+openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
+openssl req -new -key device_ec_key.pem -out device_ec.csr -subj "/CN=${1}"
```

Comment:
>1 [](start = 72, length = 1)

if passing parameters please use following bash script checks

set -o errexit # Exit if command failed.
set -o nounset # Exit if variable not set.
set -o pipefail # Exit if pipe failed.

Reply:
example
[https://expresslogic.visualstudio.com/X-Ware/_git/azure-services-port?path=%2Fsdk%2Ftests%2Fe2e%2Fservice%2Fmocha_exec.sh](https://expresslogic.visualstudio.com/X-Ware/_git/azure-services-port?path=%2Fsdk%2Ftests%2Fe2e%2Fservice%2Fmocha_exec.sh)

---
In reply to: [420364879](https://github.com/Azure/azure-sdk-for-c/pull/682#discussion_r420364879) [](ancestors = 420364879)

Reply:
N/A now that it uses self-signed certs